harness/determined/profiler.py

```diff
@@ -223,16 +222,120 @@ def pop_until_deadline(q: queue.Queue, deadline: float) -> Iterator[Any]:
             break
 
 
-def profiling_metrics_exist(master_url: str, trial_id: str) -> bool:
+class TrialProfilerMetricsBatch:
+    """
+    TrialProfilerMetricsBatch is the representation of a batch of trial
+    profiler metrics as accepted by POST /api/v1/trials/:trial_id/profiler/metrics
+    """
+
+    def __init__(
+        self,
+        values: List[float],
+        batches: List[int],
+        timestamps: List[str],
+        labels: Dict[str, Any],
+    ):
+        self.values = values
+        self.batches = batches
+        self.timestamps = timestamps
+        self.labels = labels
+
+
+def post_trial_profiler_metrics_batches(
+    sess: api.Session,
+    batches: List[TrialProfilerMetricsBatch],
+) -> None:
+    """
+    Post the given metrics to the master to be persisted. Labels
+    must contain only a subset of the keys: trial_id,  name,
+    gpu_uuid, agent_id and metric_type, where metric_type is one
+    of PROFILER_METRIC_TYPE_SYSTEM or PROFILER_METRIC_TYPE_TIMING.
+    """
+    backoff_interval = 1
+    max_tries = 2
+    tries = 0
+
+    while tries < max_tries:
```

Comment:
Not your job, here, but it's weird to re-implement `urllib3` retry logic here.

Reply:
yeah, I don't fix ancient code that is destined for deletion.  That's like sanding your firewood.